lib/components/UploadPicker.vue

```diff
@@ -294,16 +294,27 @@ export default defineComponent({
 		},
 
 		hasFailure(): boolean {
-			return this.queue?.filter((upload: Upload) => upload.status === UploadStatus.FAILED).length !== 0
+			return this.queue?.some((upload: Upload) => upload.status === UploadStatus.FAILED)
 		},
 		isUploading(): boolean {
 			return this.queue?.length > 0
 		},
-		isAssembling(): boolean {
-			return this.queue?.filter((upload: Upload) => upload.status === UploadStatus.ASSEMBLING).length !== 0
+		isOnlyAssembling(): boolean {
+			return !this.queue?.some((upload: Upload) => {
+				// ignore empty uploads or meta uploads
+				if (upload.size === 0) {
+					return false
+				}
+				// If all the uploads are assembling or finished, the ongoing task is assembling
+				return upload.status !== UploadStatus.ASSEMBLING && upload.status !== UploadStatus.FINISHED
+			})
 		},
 		isPaused(): boolean {
-			return this.uploadManager.info?.status === Status.PAUSED
+			return this.uploaderStatus === Status.PAUSED
+		},
+
+		uploaderStatus(): Status {
+			return this.uploadManager.info?.status || Status.IDLE
```

Comment:
```suggestion
			return this.uploadManager.info?.status ?? Status.IDLE
```

Reply:
But it can not even be undefined. Adding the fallback nevertheless will make real bugs hard to spot.

```suggestion
			return this.uploadManager.info.status
```